drivers/wireless/ieee80211/bcm43xxx/bcmf_sdio.c

```diff
@@ -1077,17 +1078,12 @@ struct bcmf_sdio_frame *bcmf_sdio_allocate_frame(FAR struct bcmf_dev_s *priv,
           break;
         }
 
-      if (block)
-        {
-          /* TODO use signaling semaphore */
+      nxsig_usleep(10 * 1000);
 
-          wlinfo("alloc failed %d\n", tx);
-          nxsig_usleep(100 * 1000);
-          continue;
+      if (!block)
```

Comment:
why move block check after sleep

Reply:
this is a small improvement, if there is no buffer available from freequeue, sdio thread will delay for a period of time to yield the slice and switch to workqueue to process tx/rx queue further.

Reply:
if so, why not keep the frame allocation at line 1028 in the block mode?